BSP development guide.md

```diff
@@ -0,0 +1,142 @@
+# Board Support Package development guide
+
+This guide describes necessary steps add a new BSP to the esp-bsp project.
+
+### New BSP requirements
+* We want the BSPs to be used by as many users and projects as possible. All BSP must support IDF from version 4.4 up to latest release. Boards that were released after IDF v5.0 do not have to be backward compatible with v4.4.
+* We want the BSPs to provide a 'board abstraction layer' that allows the user to run his application on a range of different boards. Public API for each BSP must be the same. The API consistency is ensured by conventions described below.
+* We want the main features to be clearly visible for new users. Each new BSP must be added to a table of supported boards in root [README.md](https://github.com/espressif/esp-bsp/blob/master/README.md) it must contain a license (Apache-2) and a README.md file. Conventions for the readme file are described below
+* All BSPs are uploaded to [ESP Registry](https://components.espressif.com/). So each BSP must contain `idf_component.yml` file and must be added to [CI upload job](https://github.com/espressif/esp-bsp/blob/master/.github/workflows/upload_component.yml). The `idf_component.yml` must contain: version, description, url, target and (optional) dependencies.
+* We want to provide LVGL users with easy, out-of-the-box working solution for their GUI applications. Each board that has a display must be added to LVGL's [SquareLineStudio](https://squareline.io/) pre-defined boards. For more information refer to [this readme](https://github.com/espressif/esp-bsp/blob/master/SquareLine/README.md).
+    * List of currently implemented LCD drivers is maintained in a [separate table](https://github.com/espressif/esp-bsp/blob/master/LCD.md)
+* We have an experimental idf.py extension, [bsp_ext.py](https://github.com/espressif/esp-bsp/blob/master/examples/bsp_ext.py) that allows changing of BSP during project configuration. Please add new BSPs to the list in this file. 
+
+### API conventions
+
+BSP is a standard IDF component. You can start development by `idf.py create-component your_bsp_name`.
+
+The APIs that a BSP can offer depend on the boards capabilities. For example, a board that contains audio codec will expose audio API and a board with display and touch controller will expose display API.
```

Comment:
Missing mention of touch API at the end of this paragraph: "will expose display and **touch** API."